### PR TITLE
[navigation]fix: redirect user to home in global when workspace is enabled

### PR DESCRIPTION
**Pull request by SuZhou-Joe**
### Description

When workspace is enabled, users are not supposed to create any objects in global, thus the new left navigation need to redirect users to home, and in home, user won't be able to expand the left navigation.

- Change the `back` to `home` when user is in settings and setup / data administration nav groups.
- Change the navGroup to dataAdministration when navigate to devTools.

### Before fix

Before the fix, you can see users is able to go into all use case and create objects in global.

https://github.com/user-attachments/assets/735def6e-f363-4580-b131-40047e9816bb

### After fix

User will be redirect to home.

https://github.com/user-attachments/assets/e6360850-a431-4239-ae08-78aaa72db705

## Testing the changes

- Enable workspace feature flag
- Goes to settings and setup
- Expand the left navigation

## Changelog

- fix: [navigation]feat: redirect user to home in global when workspace is enabled

### Check List

- [x] All tests pass
  - [x] `yarn test:jest`
  - [x] `yarn test:jest_integration`
- [x] New functionality includes testing.
- [x] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/dev_tools/public/dev_tools_icon.test.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ */
+
+import React from 'react';
+import { fireEvent, render } from '@testing-library/react';
+import { DevToolsIcon } from './dev_tools_icon';
+import { coreMock } from '../../../core/public/mocks';
+import { DEFAULT_NAV_GROUPS } from '../../../core/public';
+
+describe('<DevToolsIcon />', () => {
+  it('should call some methods from core service when click', () => {
```
Comment (ruanyl) [inline src/plugins/dev_tools/public/dev_tools_icon.test.tsx:13]: Nit: better to elaborate what's the `some methods`

File: src/core/public/chrome/ui/header/collapsible_nav_group_enabled_top.tsx [hunk 2]
```diff
@@ -22,46 +22,59 @@ import { ALL_USE_CASE_ID } from '../../../../../core/utils';
...
   const homeLink = useMemo(() => navLinks.find((link) => link.id === 'home'), [navLinks]);
 
-  const shouldShowBackButton = useMemo(
-    () =>
-      currentNavGroup?.id !== ALL_USE_CASE_ID &&
-      !shouldShrinkNavigation &&
-      Object.values(navGroupsMap).filter((item) => !item.type).length > 1 &&
-      currentNavGroup,
-    [navGroupsMap, currentNavGroup, shouldShrinkNavigation]
+  const isInGlobalWorkspace = useMemo(
```
Comment (ruanyl) [inline src/core/public/chrome/ui/header/collapsible_nav_group_enabled_top.tsx:46]: What is global workspace referring? Shall we not use the term "global workspace" as we no longer have this?

**Review by ruanyl — APPROVED** [review]

File: src/core/public/chrome/ui/header/collapsible_nav_group_enabled_top.tsx (hunk 2 above)
Comment (SuZhou-Joe) [reply src/core/public/chrome/ui/header/collapsible_nav_group_enabled_top.tsx]: Sure, change to isOutsideWorkspace.

File: src/plugins/dev_tools/public/dev_tools_icon.test.tsx (hunk 1 above)
Comment (SuZhou-Joe) [reply src/plugins/dev_tools/public/dev_tools_icon.test.tsx]: Sure, updated.

**Review by ruanyl — APPROVED** [review]

**Review by ananzh — APPROVED** [review]

**Pull request merged**